### PR TITLE
Remove primary_mirror_mode stuff. [for walreplication branch]

### PR DESCRIPTION
**Pull request by hlinnaka**
Revert the state machine and other logic in postmaster.c the way it is in
upstream. Remove some GUCs related to mirrored and non-mirrored mode. Remove
the -M, -x and -y postmaster options, and change management scripts to not
pass those options.

### CODE REVIEW

**Comment by ashwinstar** [comment]
Sure thanks for taking this up, will review it today. Please can we get it GREEN here in PR pipeline before merging to branch.

File: gpMgmt/bin/gppylib/commands/gp.py [hunk 1]
```diff
@@ -371,7 +367,7 @@ def __init__(self, name, gpdb, numContentsInCluster, era, mirrormode,
 
         # build backend options
         b = PgCtlBackendOptions(port, dbid, numContentsInCluster)
-        b.set_segment(mirrormode, content)
+        b.set_segment(content)
```
Comment (cjcjameson) [inline gpMgmt/bin/gppylib/commands/gp.py]: This removal leads to some more dead code; I'll add a commit to this PR

File: src/test/walrep/expected/missing_xlog.out [hunk 2]
```diff
@@ -9,7 +9,7 @@ returns text as $$
     if command in ('stop', 'restart'):
         cmd = cmd + '-w -m immediate %s' % command
     elif command == 'start':
-        opts = '-p %d -\-gp_dbid=0 -\-silent-mode=true -i -M mirrorless -\-gp_contentid=%d -\-gp_num_contents_in_cluster=3' % (port, contentid)
+        opts = '-p %d -\-gp_dbid=0 -\-silent-mode=true -i -\-gp_contentid=%d -\-gp_num_contents_in_cluster=3' % (port, contentid)
```
Comment (cjcjameson) [inline src/test/walrep/expected/missing_xlog.out]: going to add a commit for two test cases in the isolation2 suite

File: src/backend/postmaster/postmaster.c [hunk 3]
```diff
@@ -2263,32 +1792,39 @@ ServerLoop(void)
...
+		checkPgDir("/pg_utilitymodedtmredo");
+		checkPgDir("/pg_distributedlog");
+		checkPgDir("/pg_distributedxidmap");
+		checkPgDir("/pg_multixact");
+		checkPgDir("/pg_subtrans");
+		checkPgDir("/pg_xlog");
+		checkPgDir("/pg_clog");
+		checkPgDir("/pg_multixact/members");
+		checkPgDir("/pg_multixact/offsets");
+		checkPgDir("/pg_xlog/archive_status");
```
Comment (ashwinstar) [inline src/backend/postmaster/postmaster.c]: Do we really need to check for this file `/pg_xlog/archive_status` ?? what's its used for ?

File: contrib/gp_inject_fault/gp_inject_fault.c [hunk 4]
```diff
@@ -52,102 +115,47 @@ gp_inject_fault(PG_FUNCTION_ARGS)
...
+		 * ignore it.
+		 *
+		 * (Perhaps this function should be defined as EXECUTE ON SEGMENTS,
+		 * instead of dispatching manually here? But then it wouldn't run on
+		 * QD. There is no EXECUTE ON SEGMENTS AND MASTER options, at the
+		 * moment...)
+		 */
+		char *sql;
+
+		sql = psprintf("select gp_inject_fault(%s, %s, %s, %s, %s, %d, %d, %d)",
```
Comment (ashwinstar) [inline contrib/gp_inject_fault/gp_inject_fault.c]: We just need to be aware that with newer mechanism now doesn't allow to set any follow-up faults if one is already set in executor path or maybe transaction layer as this `Select` query would itself hit the same. Earlier mechanism not going through allowed to easily set the same without worrying about it.

**Comment by jchampio** [comment]
> Please can we get it green on PR pipeline fully before pushing in.

Looks like this branch will need to be rebased over the latest `walreplication` to pick up the conan changes; the PR pipeline got stuck on the Ubuntu compilation.

**Comment by hlinnaka** [comment]
Rebased.

File: src/test/walrep/expected/missing_xlog.out (hunk 2 above)
Comment (hlinnaka) [reply src/test/walrep/expected/missing_xlog.out]: Thanks, fixed those calls in isolation2.

File: contrib/gp_inject_fault/gp_inject_fault.c (hunk 4 above)
Comment (hlinnaka) [reply contrib/gp_inject_fault/gp_inject_fault.c]: Good point. Added a comment about that, I think that will do for now.

File: src/backend/postmaster/postmaster.c (hunk 3 above)
Comment (hlinnaka) [reply src/backend/postmaster/postmaster.c]: It's used for log shipping with `archive_command`. We don't use or support that in GPDB, so I guess we wouldn't need to check for it, but this is upstream code..

Comment (hlinnaka) [reply src/backend/postmaster/postmaster.c]: I'll take that back; this is code is not in the upstream. So I'll go and remove the check for archive_status.

**Comment by hlinnaka** [comment]
The `walreplication` branch was pushed to `master`. I'll close and re-open this PR against 'master`.

**Pull request closed without merging**